lib/fs.js

```diff
@@ -1903,16 +1932,16 @@ function ReadStream(path, options) {
 
   if (this.start !== undefined) {
     if (typeof this.start !== 'number') {
-      throw new TypeError('"start" option must be a Number');
+      throw new errors.TypeError('ERR_INVALID_OPT_VALUE', 'start', this.start);
     }
     if (this.end === undefined) {
       this.end = Infinity;
     } else if (typeof this.end !== 'number') {
-      throw new TypeError('"end" option must be a Number');
+      throw new errors.TypeError('ERR_INVALID_OPT_VALUE', 'end', this.end);
     }
 
     if (this.start > this.end) {
-      throw new Error('"start" option must be <= "end" option');
+      throw new errors.Error('ERR_INVALID_OPT_VALUE', 'end', this.end);
```

Comment:
IMHO `ERR_VALUE_OUT_OF_RANGE` might be better:
```js
     throw new errors.RangeError('ERR_VALUE_OUT_OF_RANGE',
                                 'start',
                                 '<= "end"',
                                 `{start: ${this.start}, end: ${this.end}}`);
```
to produce:
`The value of "start" must be <= "end". Received "{start: 5, end: 4}"`

Reply:
I updated it to `ERR_VALUE_OUT_OF_RANGE` and I added the new error `ERR_VALUE_OUT_OF_RANGE` in `errors.js`